trunner/test_runner.py

```diff
@@ -191,17 +193,16 @@ def run_tests(self, tests: Sequence[TestOptions]):
             def set_reboot_flag(tests, idx, result):
                 # If the test is successful and the next test doesn't require loading via
                 # the plo we don't want to reboot the entire device (to speed up the test execution).
-                # There are three exceptions to this rule:
+                # There are four exceptions to this rule:
+                # 0. The yaml config has been configured with the reboot parameter set to True.
                 # 1. Runner runs in the "nightly" mode when we are not concerned about slow execution.
                 # 2. The test has failed.
                 # 3. We have to enter the bootloader in order to load applications.
                 if idx == len(tests) - 1:
                     return
 
-                tests[idx + 1].should_reboot = False
-
                 if result.is_skip():
-                    tests[idx + 1].should_reboot = tests[idx].should_reboot
+                    tests[idx + 1].reboot = tests[idx].reboot
```

Comment:
this is not entirely correct right now - IMHO the explicit "reboot before the test" property should not be forwarded to the next test on SKIP (I assume previously it might have been needed if we skipped the test which loaded necessary binary? - I'm not sure - maybe it's invalid altogether)

Reply:
I think the initial purpose of this line was passing `should_reboot` to next test when there was a fail and next skip. IMHO we should check whether there was a fail before and only for this case do it.